packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ClientMethodTemplate.java

```diff
@@ -1625,7 +1625,7 @@ private String getPagingNextPageExpression(ClientMethod clientMethod, String met
         if (settings.isDataPlaneClient() && settings.isPageSizeEnabled()) {
             Optional<String> maxPageSizeSerializedName
                 = MethodUtil.serializedNameOfMaxPageSizeParameter(clientMethod.getProxyMethod());
-            if (maxPageSizeSerializedName.isPresent() && settings.isBranded()) {
+            if (maxPageSizeSerializedName.isPresent()) {
```

Comment:
Does unbranded support maxPageSize as well?

Reply:
For unbranded, this should be the query parameter marked with `@pageSize` decorator. And the input would be in `PagingOptions` and the impl in generated code would be different.

Therefore I think it be safe to assume the `maxpagesize` query param is only for Azure.